src/handwritten/system/_collections/device_collection.py

```diff
@@ -59,6 +59,12 @@ def __getitem__(self, index):
             return [_DeviceAlternateConstructor(name, self._interpreter) for name in self.device_names[index]]
         elif isinstance(index, str):
             device_names = unflatten_channel_string(index)
+            all_devices = self.device_names
+            # Validate the device names we were provided
+            for device in device_names:
+                if device not in all_devices:
+                    raise KeyError(f'"{device}" is not a valid device name.')
```

Comment:
This validation doesn't have the same problem with internal names that the physical channel validation does, but we may need to add internal C APIs to do this validation anyway: AB#2663510